cmd/ethereum_rust/ethereum_rust.rs

```diff
@@ -118,23 +118,33 @@ async fn main() {
     store
         .add_initial_state(genesis.clone())
         .expect("Failed to create genesis block");
-
     if let Some(chain_rlp_path) = matches.get_one::<String>("import") {
         let blocks = read_chain_file(chain_rlp_path);
         let size = blocks.len();
-        for block in blocks {
+        for block in &blocks {
             let hash = block.header.compute_block_hash();
             info!(
                 "Adding block {} with hash {:#x}.",
                 block.header.number, hash
             );
-            if add_block(&block, &store).is_err() {
+            if add_block(block, &store).is_err() {
                 warn!(
                     "Failed to add block {} with hash {:#x}.",
                     block.header.number, hash
                 );
             }
         }
+        if let Some(last_block) = &blocks.last() {
```

Comment:
calling apply fork choice function should do the trick.

Reply:
It works!